pkgs/development/python-modules/ratarmountcore/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, isPy3k
+, fusepy
+, indexed_bzip2
+, indexed_gzip
+, indexed_zstd
+, lzmaffi
+, python-xz
+, rapidgzip
+, rarfile
+}:
+
+buildPythonPackage rec {
+  pname = "ratarmountcore";
+  version = "0.6.0";
+
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "564393eae729a6fba0ed6964dc5a47d1a4aeabedd3b486a6453bb859a4f9c499";
+  };
+
+  propagatedBuildInputs = [ indexed_gzip indexed_bzip2 indexed_zstd fusepy lzmaffi python-xz rapidgzip rarfile ];
+
+  pythonImportsCheck = [ "ratarmountcore" ];
+
+  doCheck = false; # unclear cause of errors
```

Comment:
I looked into this a bit. Using the [pytestCheckHook](pytestCheckHook) gets the tests running.

However, most of the tests require test articles using a script bundled with ratarcore.
ratarcore calls these tests and does some of its own from what I understand?
So I would just update this comment to `# tests covered by ratarmount` or something

Since you are the author, you might want to think about changing this upstream at some point if you want to decouple them, but I don't think it's that much of a problem.

Reply:
Both, ratarmountcore and ratarmount have some tests. The tests in ratarmount are definitely nothing to run on a user system. They take an hour and require extra files. It's a simple bash script iterating over many regression tests. Ratarmountcore has some tests that can be run with pytest but some of the tests require test files that are only in the git repository and not distributed to PyPI because they would only be ballast.

I played around a bit and got the rest of the tests for ratarmountcore to work. I'll push the changes.

Reply:
The output of `nix-build -A ratarmount` contains:
```python3
adding 'ratarmountcore-0.6.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
installing
Executing pypaInstallPhase
Successfully installed ratarmountcore-0.6.0-py3-none-any.whl
Finished executing pypaInstallPhase
pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/f5jppcxva77ib59040ly7qpkvdwvssnq-python3.10-ratarmountcore-0.6.0
checking for references to /build/ in /nix/store/f5jppcxva77ib59040ly7qpkvdwvssnq-python3.10-ratarmountcore-0.6.0...
patching script interpreter paths in /nix/store/f5jppcxva77ib59040ly7qpkvdwvssnq-python3.10-ratarmountcore-0.6.0
stripping (with command strip and flags -S -p) in  /nix/store/f5jppcxva77ib59040ly7qpkvdwvssnq-python3.10-ratarmountcore-0.6.0/lib
shrinking RPATHs of ELF executables and libraries in /nix/store/h9hdkpscfhyii43kdmdjr8fnrkl450y6-python3.10-ratarmountcore-0.6.0-dist
checking for references to /build/ in /nix/store/h9hdkpscfhyii43kdmdjr8fnrkl450y6-python3.10-ratarmountcore-0.6.0-dist...
patching script interpreter paths in /nix/store/h9hdkpscfhyii43kdmdjr8fnrkl450y6-python3.10-ratarmountcore-0.6.0-dist
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-7.4.0, pluggy-1.2.0
rootdir: /build/ratarmountcore-0.6.0
collected 310 items                                                            

tests/test_BlockParallelReaders.py ..................................... [ 11%]
........................................................................ [ 35%]
........................................................................ [ 58%]
........................................................................ [ 81%]
.............                                                            [ 85%]
tests/test_SQLiteBlobFile.py ..                                          [ 86%]
tests/test_SQLiteIndex.py .                                              [ 86%]
tests/test_StenciledFile.py .......................                      [ 94%]
tests/test_UnionMountSource.py ....                                      [ 95%]
tests/test_compressions.py ....                                          [ 96%]
tests/test_factory.py ...                                                [ 97%]
tests/test_utils.py .......                                              [100%]

=============================== warnings summary ===============================
../../nix/store/9q5q1l8czb41jff9vpnsbfn0pg23ljf1-python3.10-cffi-1.15.1/lib/python3.10/site-packages/cffi/vengine_cpy.py:4
  /nix/store/9q5q1l8czb41jff9vpnsbfn0pg23ljf1-python3.10-cffi-1.15.1/lib/python3.10/site-packages/cffi/vengine_cpy.py:4: DeprecationWarning: the imp module is deprecated in favour of importlib and slated for removal in Python 3.12; see the module's documentation for alternative uses
    import sys, imp

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================= 310 passed, 1 warning in 38.27s ========================
installCheckPhase completed in 39 seconds
pythonCatchConflictsPhase
/nix/store/91qrq1mz3d6qz4lv2zirswb3z5qr538l-catch_conflicts.py:1: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: ratarmountcore
pytestcachePhase
pytestRemoveBytecodePhase
building '/nix/store/pdn5dwy8bnhpfj15zh9fs7zwqw7gr9m2-python3.10-ratarmount-0.14.0.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/11g6s7m8wyrbqzh1lrnzhhn3gjf0vkya-ratarmount-0.14.0.tar.gz
source root is ratarmount-0.14.0
```

Better than nothing, I guess.